apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx

```diff
@@ -119,73 +86,133 @@ export const LBPForm = ({ pool }: Props) => {
   };
 
   const createChangeHandler = (fieldKey: keyof Values) => (event: React.ChangeEvent<any>) => {
-    setFormValue(fieldKey, event.target.value);
+    let value = event.target.value;
+
+    if (['startWeight1', 'startWeight2', 'endWeight1', 'endWeight2'].indexOf(fieldKey) >= 0) {
+      value = DecimalBigNumber.parseUnits(value, 16);
+    }
+
+    if (['startDate', 'endDate'].indexOf(fieldKey) >= 0) {
+      try {
+        value = parse(value, "yyyy-LL-dd'T'kk:mm", new Date());
+      } catch (error) {
+        console.error(error);
+        value = new Date();
+      }
+    }
+
+    setFormValue(fieldKey, value);
   };
 
   const saveForm = () => {
-    console.log('Saving form, with values');
-    console.log(formValues);
+    return createPool.handler({
+      name: formValues.name,
+      symbol: formValues.symbol,
+      tokenAddresses: [formValues.releasedToken, formValues.accruedToken],
+      feePercentage: formValues.fees,
+      swapEnabledOnStart: true,
+      weights: [formValues.startWeight1, formValues.startWeight2],
+    });
   };
 
-  const completePoolSetup = () => {
-    console.log('Complete pool setup');
-    console.log('Invoke upgradeWeightsGradually');
+  const updateWeights = () => {
+    return updateWeightsGradually.handler(
+      formValues.startDate,
+      formValues.endDate,
+      formValues.endWeight1,
+      formValues.endWeight2
+    );
   };
 
-  const pausePool = () => {
-    console.log('Invoke pause pool');
+  const updateSwapEnabled = async (enabled: boolean) => {
+    return setSwapEnabled.handler(enabled);
   };
 
   const drainPool = () => {
     console.log('Invoke drain pool');
   };
 
-  // TODO: Do we need any of the other fields in the form? I suspect we do.
+  const formatDate = (date: Date) => {
+    if (!date) return;
+    let result;
+    try {
+      result = format(date, "yyyy-LL-dd'T'kk:mm");
+      return result;
+    } catch (error) {
+      console.error(error);
+      result = format(new Date(), "yyyy-LL-dd'T'kk:mm");
+    }
+    return result;
+  };
+
+  const formatWeight = (weight: DecimalBigNumber) => {
+    if (!weight) return '0';
+    return Math.trunc(Number(formatUnits(weight.value, 16)));
```

Comment:
DecimalBigNumber has the `formatUnits` as part of the class -- can just do 
```
Math.trunc(Number(weight.formatUnits()))
```

Reply:
I'm not sure why but that isn't giving me the expected result. Unless you strongly object we can leave it as is and I can work on it in another PR.